### PR TITLE
homework - 2

### PR DESCRIPTION
**Pull request by Artemijacd**

### CODE REVIEW

File: docs/3-building-blocks-of-oop-part-2/hometask/src/Item.ts [hunk 1]
```diff
@@ -3,9 +3,70 @@ import { Comparable } from './Comparable';
 let id = 0;
 
 export abstract class Item implements Comparable<Item> {
+    
+    private id: number;
```
Comment (metam0rphus) [inline docs/3-building-blocks-of-oop-part-2/hometask/src/Item.ts:7]: private readonly id
a bit better

File: docs/3-building-blocks-of-oop-part-2/hometask/src/Item.ts [hunk 2]
```diff
@@ -3,9 +3,70 @@ import { Comparable } from './Comparable';
...
+    private value: number;
+    private weight: number;
+    private name: string;
+
+
+    constructor(name: string, value: number, weight: number);
```
Comment (metam0rphus) [inline docs/3-building-blocks-of-oop-part-2/hometask/src/Item.ts:13]: this is unnecessary :)

File: docs/3-building-blocks-of-oop-part-2/hometask/src/Item.ts [hunk 3]
```diff
@@ -3,9 +3,70 @@ import { Comparable } from './Comparable';
...
+    constructor(name: string, value: number, weight: number) {
+        this.setNumberOfItems();
```
Comment (metam0rphus) [inline docs/3-building-blocks-of-oop-part-2/hometask/src/Item.ts:15]: better to do this
        this.id = id++;
        this.name = name;
        this.value = value;
        this.weight = weight;

methods can be overridden in descendants which can break the state
this.setNumberOfItems();
this.setName(name)
this.setValue(value);
this.setWeight(weight)

File: docs/3-building-blocks-of-oop-part-2/hometask/src/Item.ts [hunk 4]
```diff
@@ -3,9 +3,70 @@ import { Comparable } from './Comparable';
...
+    public setName(name: string): void {
+        this.name = name;
+    }
+
+    public setWeight(weight: number): void {
+        this.weight = weight;
     }
 
-    // your code goes here
+    public setNumberOfItems(): void {
```
Comment (metam0rphus) [inline docs/3-building-blocks-of-oop-part-2/hometask/src/Item.ts:65]: this is unnecessary or it should be private

File: docs/3-building-blocks-of-oop-part-2/hometask/src/ItemWeightComparator.ts [hunk 5]
```diff
@@ -2,7 +2,7 @@ import { Item } from './Item';
 import { ItemComparator } from './ItemComparator';
 
 export class ItemWeightComparator implements ItemComparator {
-    public compare(first: Item, second: Item) {
-        // your code goes here
+    public compare(first: Item, second: Item): number {
+        return first.getWeight() - second.getWeight();
```
Comment (metam0rphus) [inline docs/3-building-blocks-of-oop-part-2/hometask/src/ItemWeightComparator.ts:6]: If the weights are equal, this method should call the compareTo(other: Item) method of the first Item and return the resulting value.

File: docs/3-building-blocks-of-oop-part-2/hometask/src/Sword.ts [hunk 6]
```diff
@@ -1 +1,21 @@
-// your code goes here
\ No newline at end of file
+import { Weapon } from "./Weapon";
+export class Sword extends Weapon {
```
Comment (metam0rphus) [inline docs/3-building-blocks-of-oop-part-2/hometask/src/Sword.ts:2]: Best solution. Try to think it over
export class Sword extends Weapon {
    private readonly maxDamage: number;

    constructor(baseDamage: number, baseDurability: number, value: number, weight: number) {
        super('sword', baseDamage, baseDurability, value, weight);
        this.maxDamage = baseDamage * 1.25;
    }

    polish(): void {
        if (this.getDamage() + Sword.MODIFIER_CHANGE_RATE > this.maxDamage) {
            this.setDamage(this.maxDamage);
        }
        this.setDamage(this.getDamage() + Sword.MODIFIER_CHANGE_RATE);
    }
}

File: docs/3-building-blocks-of-oop-part-2/hometask/src/Bow.ts [hunk 7]
```diff
@@ -1 +1,18 @@
-// your code goes here
\ No newline at end of file
+import { Weapon } from "./Weapon";
+export class Bow extends Weapon {
```
Comment (metam0rphus) [inline docs/3-building-blocks-of-oop-part-2/hometask/src/Bow.ts:2]: the same as sword

File: docs/3-building-blocks-of-oop-part-2/hometask/src/Inventory.ts [hunk 8]
```diff
@@ -1 +1,29 @@
...
+    public addItem(item: Item): void {
+        this.items.push(item);
+    }
+
+    sort(): void;
+    sort(comparator: ItemWeightComparator): void;
+
+    public sort(comparator?: ItemWeightComparator): void {
+        if (comparator === undefined) {
+            this.items.sort((a, b) => a.getValue() - b.getValue());
```
Comment (metam0rphus) [inline docs/3-building-blocks-of-oop-part-2/hometask/src/Inventory.ts:20]: Item has compareTo

File: docs/3-building-blocks-of-oop-part-2/hometask/src/Pizza.ts [hunk 9]
```diff
@@ -1 +1,27 @@
...
+    private slicesEaten: number;
+
+    constructor(numberOfSlices: number, spoiled: boolean)
+    constructor(numberOfSlices: number, spoiled: boolean) {
+        super('pizza', numberOfSlices*2, numberOfSlices * 0.05, spoiled)
+        this.numberOfSlices = numberOfSlices;
+        this.slicesEaten = 0;
+    }
+
+    public eat(): void {
```
Comment (metam0rphus) [inline docs/3-building-blocks-of-oop-part-2/hometask/src/Pizza.ts:13]: You should check isSpoiled as well

File: docs/3-building-blocks-of-oop-part-2/hometask/src/Item.ts (hunk 1 above)
Comment (Artemijacd) [reply docs/3-building-blocks-of-oop-part-2/hometask/src/Item.ts]: fixed

**Pull request merged**